Overhaul powa-web documentation

rjuju opened the pull request:
Add a new install section and do a major rewrite of the installation documentation.

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+Introduction
+------------
+
+PoWA-web is the dedicated user interface of the PoWA project.  It gathers the
+data on a wanted time interval from the repository server and produces graphs,
+grids and various optimisation suggestions.
```
```suggestion
PoWA-web is the dedicated user interface of the PoWA project.  It produces graphs,
grids and various optimization suggestions using the data from the repository
server.
```

rjuju replied:
thanks, I retweaked it a bit more.

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+***********************************************
+
+Please refer to the `APT PGDG repository documentation
+<https://apt.postgresql.org>` for the initial setup (which should already be
+done if you have PostgreSQL installed), and simply install
```
```suggestion
<https://apt.postgresql.org>` for the initial setup (which might already be
done as this is a convenient Debian PostgreSQL repository), and simply install
```

Maybe add a a footnote to https://wiki.postgresql.org/wiki/Apt/FAQ#What.27s_the_difference_between_the_PostgreSQL_packages_provided_by_Debian.2FUbuntu_and_by_the_PGDG_apt_repository.3F ?

rjuju replied:
This part is still in the context of this documentation, so if you installed powa-archivist following the PGDG repository documentation and then install powa-web using the same method then yes the initial setup should have been done.

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+.. code-block:: bash
+
+  sudo apt-get install powa-web
+
```
```suggestion

Then, jump to the configuration chapter.
```

It would be cool to add internal referential links…

rjuju replied:
You mean a link to literally the next line?

ioguix replied:
No, the next line is the "Manuel installation" method chapter, this later ending with the same sentence where the "jump" is really tiny over there :)

rjuju replied:
oh the "Then, jump to the configuration chapter." is a new one, I was looking at the existing one.  Sorry the github interface is so confusing and I'm not familiar with the suggested change thing.

rjuju replied:
I added the "then jump" sentence with a link to the configuration section.

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+* ~/.powa-web.conf
+* ./powa-web.conf
+
+You'll then be noticed of the address and port on which the UI is available.
```
Maybe this is a log message after starting powa-web?

```suggestion
After starting powa-web, you'll find the address and port on which the UI is available in log messages.
```

rjuju replied:
also tweaked it a bit and moved to the deployment part.

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+* ./powa-web.conf
+
+You'll then be noticed of the address and port on which the UI is available.
+The default is 0.0.0.0:8888, as indicated in this message:
```
```suggestion
The default is 0.0.0.0:8888, as indicated in this example:
```

ioguix commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+The following options are required:
+
+servers (dict):
+  A dictionary mapping server names to connection information.
```
```suggestion
  A dictionary mapping PostgreSQL repository servers names to their connection informations.
```

rjuju replied:
I tweaked that a bit.

The terminology in the documentation is "PoWA repository server", which implies that it's a postgres server too.

Note also that in english "information" is an uncountable noun so it's not used in a plural form.

rjuju commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+* ~/.powa-web.conf
+* ./powa-web.conf
+
+You'll then be noticed of the address and port on which the UI is available.
+The default is 0.0.0.0:8888, as indicated in this message:
+
+* [I 161105 20:27:39 powa-web:12] Starting powa-web on 0.0.0.0:8888
```
I had the same idea when looking at it.  I didn't notice it as first as this part has been there for quite some time.

I reworded it and moved it.  It also reminds me that I added support for native SSL connections but it's not documented yet, I will take care of that once this PR is merged.

rjuju replied:
actually I meant the deployment section, as this is where the `powa-web` script is mentioned.

pgiraud commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,188 @@
+Introduction
+------------
+
+PoWA-web is the dedicated user interface of the PoWA project.  It gathers the
```
There’s a double space before “it”.

rjuju replied:
yes it's on purpose, and I usually do that kind of everywhere.

I got used to it in upstream postgres sources, and after some time I did appreciate it as I think it makes the source more readable.

rjuju commented:
Force pushed with the modifications.

pgiraud commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,192 @@
+You'll then be notified of the address and port on which the UI is available.
+The default is 0.0.0.0:8888, as indicated in this message:
+
+* [I 161105 20:27:39 powa-web:12] Starting powa-web on 0.0.0.0:8888
```
```
.. code-block::

    [I 161105 20:27:39 powa-web:12] Starting powa-web on https://0.0.0.0:8888
```

rjuju replied:
thanks, fixed

pgiraud commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,192 @@
+based on Debian/Ubuntu.
+
+For other platforms, or if you need a different version from the one provided
+by the PGDG repositories, you can either use
```
by the PGDG repositories, you can either:

rjuju replied:
fixed

pgiraud commented on docs/components/powa-web/installation.rst:
```diff
@@ -0,0 +1,192 @@
+
+* use the `provided container images
+  <https://hub.docker.com/r/powateam/powa-web>`_,
+* install it from `pypi <https://pypi.org/project/powa-web/>`_
```
``` * install it from `pypi <https://pypi.org/project/powa-web/>`_,```

rjuju replied:
added missing trailing comma

rjuju commented:
I think I addressed all the comments.  I'm merging it as it's still a good improvement compared to the current documentation, feel free to report issues and/or submit PR if I missed something.  The documentation needs a lot of TLC anyway so there should be a lot of additional work done soon.
